metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -0,0 +1,148 @@
+// import {
+//   MySQLInstrumentation,
+//   MySQLInstrumentationConfig,
+// } from '@opentelemetry/instrumentation-mysql';
+import {
+  IORedisInstrumentation,
+  IORedisInstrumentationConfig,
+} from '@opentelemetry/instrumentation-ioredis';
+// import {
+//   MongoDBInstrumentation,
+//   MongoDBInstrumentationConfig,
+// } from '@opentelemetry/instrumentation-mongodb';
+import {
+  PgInstrumentation,
+  PgInstrumentationConfig,
+} from '@opentelemetry/instrumentation-pg';
+// import {
+//   RedisInstrumentation,
+//   RedisInstrumentationConfig,
+// } from '@opentelemetry/instrumentation-redis';
+import {
+  Instrumentation,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+
+type InstrumentationMapKeys =
+  | '@opentelemetry/instrumentation-dns'
+  | '@opentelemetry/instrumentation-express'
+  | '@opentelemetry/instrumentation-http'
+  | '@opentelemetry/instrumentation-graphql'
+  | '@opentelemetry/instrumentation-grpc'
+  | '@opentelemetry/instrumentation-koa'
+  | '@opentelemetry/instrumentation-ioredis'
+  | '@opentelemetry/instrumentation-express'
+  // | '@opentelemetry/instrumentation-mongodb'
+  // | '@opentelemetry/instrumentation-mysql'
+  | '@opentelemetry/instrumentation-pg';
+// | '@opentelemetry/instrumentation-redis'
+
+type InstrumentationConfigMap = {
+  '@opentelemetry/instrumentation-dns'?: DnsInstrumentationConfig;
+  '@opentelemetry/instrumentation-express'?: ExpressInstrumentationConfig;
+  '@opentelemetry/instrumentation-http'?: HttpInstrumentationConfig;
+  '@opentelemetry/instrumentation-graphql'?: GraphQLInstrumentationConfig &
+    InstrumentationConfig;
+  '@opentelemetry/instrumentation-grpc'?: GrpcInstrumentationConfig;
+  '@opentelemetry/instrumentation-koa'?: InstrumentationConfig;
+  '@opentelemetry/instrumentation-ioredis'?: IORedisInstrumentationConfig;
+  // '@opentelemetry/instrumentation-mongodb'?: MongoInstrumentationConfig,
+  // '@opentelemetry/instrumentation-mysql'?: MysqlInstrumentationConfig,
+  '@opentelemetry/instrumentation-pg'?: PgInstrumentationConfig;
+  // '@opentelemetry/instrumentation-redis'?: RedisInstrumentationConfig,
+};
+
+const InstrumentationMap: Record<InstrumentationMapKeys, any> = {
+  '@opentelemetry/instrumentation-dns': DnsInstrumentation,
+  '@opentelemetry/instrumentation-express': ExpressInstrumentation,
+  '@opentelemetry/instrumentation-http': HttpInstrumentation,
+  '@opentelemetry/instrumentation-graphql': GraphQLInstrumentation,
+  '@opentelemetry/instrumentation-grpc': GrpcInstrumentation,
+  '@opentelemetry/instrumentation-koa': KoaInstrumentation,
+  '@opentelemetry/instrumentation-ioredis': IORedisInstrumentation,
+  // '@opentelemetry/instrumentation-mongodb': MongoDBInstrumentation,
+  // '@opentelemetry/instrumentation-mysql': MySQLInstrumentation,
+  '@opentelemetry/instrumentation-pg': PgInstrumentation,
+  // '@opentelemetry/instrumentation-redis': RedisInstrumentation,
+};
+
+const defaultConfigs: InstrumentationConfigMap = {
```

Comment:
Maybe easier to manage a single object which has the instrumentation and the config?

```typescript
const InstrumentationMap = {
    '@opentelemetry/instrumentation-dns': {
        instrumentation: DnsInstrumentation,
        config: { enabled: true } as DnsInstrumentationConfig,
    },
    '@opentelemetry/instrumentation-express': {
        instrumentation: ExpressInstrumentation,
        config: { enabled: true } as ExpressInstrumentationConfig,
    },
    // etc
}

type InstrumentationConfigMap = {
    [Name in keyof typeof InstrumentationMap]?: typeof InstrumentationMap[Name]["config"]
};
```

Reply:
If you would like I can make a PR into your branch with these changes applied

Reply:
Tried that but it was still complaining with some type when trying to create a defaultConfig - it was more and more casting so I finally decided to not complicate it anymore to be easy to change it in future. I think it can be still refactored later if you want, WDYT ?

Reply:
Here is what i'm talking about. If you don't like it feel free to close it https://github.com/obecny/opentelemetry-js-contrib/pull/2